File: perl-programming/examples/subroutines/fibonacci.pl
#!/usr/bin/perl
use strict;
use warnings;

my @numbers = fibonacci(10);
print "@numbers\n";

sub fibonacci {
    my $num = shift;

    if ($num == 1) {
        return (1);
    } 
    if ($num == 2) {
        return (1, 1);
    }
    my @fib = (1, 1);
    foreach (3..$num) {
        push @fib, $fib[-1]+$fib[-2];
    }
    return @fib;
}

File: python/examples/functions/fibonacci.py
def fib(n):
    if n == 1:
        return 1
    if n == 2:
        return 1
    return fib(n-1) + fib(n-2)

print(3, fib(3))    # 2
print(30, fib(30))  # 832040

